src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,225 @@
+        T finVal = current.value;
+        current.value = value;
+        return finVal;
+    }
+
+    public Node<T> node(int index) {
+        if (index < (size >> 1)) {
+            Node<T> x = head;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+            return x;
+        } else {
+            Node<T> x = tail;
+            for (int i = size - 1; i > index; i--) {
+                x = x.prev;
+            }
+            return x;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        if (index == 0) {
+            final T temp = head.value;
+            head = head.next;
+            size--;
+            if (head == null) {
+                tail = null;
+            }
+            return temp;
+        } else if (index == size - 1) {
+            if (size == 0) {
+                return null;
+            } else if (size == 1) {
+                T temp = head.value;
+                head = tail = null;
+                size = 0;
+                return temp;
+            } else {
+                Node<T> current = head;
+                for (int i = 0; i < size - 2; i++) {
+                    current = current.next;
+                }
+                final T temp = tail.value;
+                tail = current;
+                tail.next = null;
+                size--;
+                return temp;
+            }
+        } else {
+            Node<T> previous = head;
+            for (int i = 0; i < index; i++) {
+                previous = previous.next;
+            }
+            Node<T> current = previous.next;
+            previous.next = current;
+            current.prev = previous.prev;
+            previous.prev.next = current;
+            T temp = previous.value;
+            size--;
+            return temp;
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> x = head; x != null; x = x.next) {
+                if (x.value == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = head; x != null; x = x.next) {
+                if (object.equals(x.value)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.next = next;
+            this.value = value;
+            this.prev = prev;
+        }
+    }
+
+    public void unlink(Node<T> current) {
```

Comment:
This and also for all methods below
```suggestion
    private void unlink(Node<T> current) {
```